internal/server/handle_test.go

```diff
@@ -0,0 +1,109 @@
+package server_test
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/drykit-go/testx"
+	"github.com/drykit-go/testx/check"
+
+	"github.com/benchttp/cobaye/internal/server"
+)
+
+func TestHandleRequest(t *testing.T) {
+	t.Cleanup(server.ResetTotalRequest)
+
+	s := server.Server{}
+
+	t.Run("request with delay param", func(t *testing.T) {
+		const delay = 100
+		const expmin, expmax = delay * time.Millisecond, (delay + 10) * time.Millisecond
+
+		r := httptest.NewRequest("", fmt.Sprintf("/?delay=%d", delay), nil)
+
+		testx.HTTPHandler(s).WithRequest(r).
+			Response(checkStatusCode(200)).
+			Duration(check.Duration.InRange(expmin, expmax)).
+			Run(t)
+	})
+
+	t.Run("request with fib param", func(t *testing.T) {
+		const fib = 35
+		const expmin, expmax = 30 * time.Millisecond, 80 * time.Millisecond
```

Comment:
Do we need to check the duration for this one?
I feel like checking that we receive a 200 status code is enough.